Source/engine/random.cpp

```diff
@@ -55,6 +55,15 @@ int32_t GenerateRnd(int32_t v)
 	return AdvanceRndSeed() % v;
 }
 
+int32_t RndBetween(int32_t a, int32_t b, bool shift)
+{
+	if (shift) {
+		a <<= 6;
+		b <<= 6;
+	}
+	return a + GenerateRnd(b - a + 1);
+}
```

Comment:
I reckon removing the shift param and requiring the two places that use it do the fixed point conversion at the call site would be less surprising. Providing the helper still improves things by letting them do `RndBetween(min << 6, max << 6);` instead of `min << 6 + GenerateRnd(max << 6 - min << 6 + 1);`.

Reply:
Gonna do that since you are a second person that suggested it :P